datafusion/physical-plan/src/aggregates/row_hash.rs

```diff
@@ -353,7 +355,7 @@ pub(crate) struct GroupedHashAggregateStream {
 
     /// scratch space for the current input [`RecordBatch`] being
     /// processed. Reused across batches here to avoid reallocations
-    current_group_indices: Vec<usize>,
+    current_group_indices: Vec<u64>,
```

Comment:
what is the reason to use u64 instead of usize

Reply:
> what is the reason to use u64 instead of usize

I think the clearer u64 is needed when we impl the blocked impl, 
we need to split the gourp_idx to two parts:
- high 32bits used to represent the `block id`
- low 32bits used to represent the `block offset`

So for reusing the same `current_group_indices` buffer in both `flat` and `blocked` mode, I modify all related `group_idx` to u64.

Reply:
Another alternative might be to ensure that the block size is aligned across the aggegators and group values -- that way there would be no stitching arrays together into batches during emission